File: 第五章_链表/链表(1)/访问单个节点的删除练习题/Remove.java
public class ListNode {
	int val;
	ListNode next = null;

	ListNode(int val) {
		this.val = val;
	}
}

public class Remove {
	public ListNode removeNode(ListNode pNode, int delVal) {
		ListNode temp = pNode;

		if (pNode.val == delVal) {// 首先判断头节点是否为要被删除节点
			pNode = pNode.next;
			return pNode;
		}

		while (temp.next != null) {// 当被删除节点存在中间部分时
			if (temp.next.val == delVal) {
				temp.next = temp.next.next;
				return pNode;
			}
			temp = temp.next;
		}

		if (temp.next == null) {// 被删除节点为尾节点时
			temp = null;
		}
		return pNode;
	}
}

File: 第五章_链表/链表(1)/访问单个节点的删除练习题/Remove.cpp
/*
struct ListNode {
    int val;
    struct ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};*/
class Remove {
public:
    ListNode* removeNode(ListNode* pHead, int delVal) {
        ListNode * temp=pHead;

		if(pHead->val==delVal){//首先判断头节点是否为要被删除节点
			pHead=pHead->next;
			return pHead;
		}

		while(temp->next!=NULL){//当被删除节点存在中间部分时
			if(temp->next->val==delVal){
				temp->next=temp->next->next;
				return pHead;
			}
			temp=temp->next;
		}

		if(temp->next==NULL){//被删除节点为尾节点时
			temp=NULL;
		}

		return pHead;
    }
};